### PR TITLE
TensorVisIO: Added VTK to Tensor Field 2D processor

### PR DESCRIPTION
**Pull request by JochenJankowai**

### CODE REVIEW

File: tensorvis/tensorvisio/include/modules/tensorvisio/processors/vtktotensorfield2d.h [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ *   * __<Outport1>__ <description>.
+ *
+ * ### Properties
+ *   * __<Prop1>__ <description>.
+ *   * __<Prop2>__ <description>
+ */
+
+/**
+ * \brief VERY_BRIEFLY_DESCRIBE_THE_PROCESSOR
+ * DESCRIBE_THE_PROCESSOR_FROM_A_DEVELOPER_PERSPECTIVE
```
Comment (martinfalk) [inline tensorvis/tensorvisio/include/modules/tensorvisio/processors/vtktotensorfield2d.h]: remove

File: tensorvis/tensorvisio/include/modules/tensorvisio/processors/vtktotensorfield2d.h [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+#include <inviwo/vtk/ports/vtkdatasetport.h>
+#include <modules/tensorvisbase/ports/tensorfieldport.h>
+
+namespace inviwo {
+
+/** \docpage{org.inviwo.VTKToTensorField2D, VTKTo Tensor Field2D}
+ * ![](org.inviwo.VTKToTensorField2D.png?classIdentifier=org.inviwo.VTKToTensorField2D)
+ * Explanation of how to use the processor.
+ *
+ * ### Inports
```
Comment (martinfalk) [inline tensorvis/tensorvisio/include/modules/tensorvisio/processors/vtktotensorfield2d.h]: update / use menu option to generate new docs

File: tensorvis/tensorvisio/src/processors/vtktotensorfield2d.cpp [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    : Processor()
+    , vtkDataSetInport_("vtkDataSetInport")
+    , tensorField2DOutport_("tensorField2DOutport") {
+    addPort(vtkDataSetInport_);
+    addPort(tensorField2DOutport_);
+}
+
+void VTKToTensorField2D::process() {
+    const auto inviwoDataSet = vtkDataSetInport_.getData();
+    const auto& vtkDataSet = *vtkDataSetInport_.getData();
```
Comment (martinfalk) [inline tensorvis/tensorvisio/src/processors/vtktotensorfield2d.cpp]: is this some convenience cast with `*`?

Comment (martinfalk) [reply tensorvis/tensorvisio/src/processors/vtktotensorfield2d.cpp]: Or what is the difference between `inviwoDataSet` and `vtkDataSet`?

File: tensorvis/tensorvisio/src/processors/vtktotensorfield2d.cpp [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+    size3_t dimensions{};
+    if (auto d = inviwoDataSet->getDimensions())
+        dimensions = *d;
+    else {
+        LogError("VTK data set type not supported.");
```
Comment (martinfalk) [inline tensorvis/tensorvisio/src/processors/vtktotensorfield2d.cpp]: should be throwing an exception instead

File: tensorvis/tensorvisio/src/processors/vtktotensorfield2d.cpp [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        return;
+    }
+    if (dimensions.z != 1) {
+        LogError("Z dimension != 1. Aborting.");
```
Comment (martinfalk) [inline tensorvis/tensorvisio/src/processors/vtktotensorfield2d.cpp]: exception

File: tensorvis/tensorvisio/src/processors/vtktotensorfield2d.cpp [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+        return;
+    }
+
+    const auto numberOfElements = glm::compMul(dimensions);
+
+    const auto pointData = vtkDataSet->GetPointData();
+    const auto tensors = pointData->GetTensors();
+
+    if (tensors->GetDataType() != VTK_FLOAT && tensors->GetDataType() != VTK_DOUBLE) {
+        LogError("Data type is neither float nor double. Aborting.");
```
Comment (martinfalk) [inline tensorvis/tensorvisio/src/processors/vtktotensorfield2d.cpp]: exception

Comment (martinfalk) [reply tensorvis/tensorvisio/src/processors/vtktotensorfield2d.cpp]: what is the reason for sticking to float/double? Is it possible to dispatch from vtk side?

File: tensorvis/tensorvisio/src/processors/vtktotensorfield2d.cpp [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+    tensorVector2D.resize(numberOfElements);
+
+    // Check to see if we need to project the tensor
+    // Assume that we project along Z axis
+    if (tensors->GetNumberOfComponents() == 9) {
+        std::vector<dmat3> tensorVector3D;
+        tensorVector3D.resize(numberOfElements);
+        auto tensorVector3DRawData = reinterpret_cast<double*>(tensorVector3D.data());
+
+        if (tensors->GetDataType() == VTK_DOUBLE) {
```
Comment (martinfalk) [inline tensorvis/tensorvisio/src/processors/vtktotensorfield2d.cpp]: dispatch would be more generic...

File: tensorvis/tensorvisio/src/processors/vtktotensorfield2d.cpp (hunk 6 above)
Comment (JochenJankowai) [reply tensorvis/tensorvisio/src/processors/vtktotensorfield2d.cpp]: Mainly two reasons:
1. The tensor field class is not templated (always stores doubles)
2. I would argue that in the case of tensors you will never encounter anything other than floats or doules.

**Pull request merged**